src/main/java/com/todo/todo/service/UserServiceImpl.java

```diff
@@ -0,0 +1,147 @@
+package com.todo.todo.service;
+
+import com.todo.todo.model.Todo;
+import com.todo.todo.model.UserRoles;
+import com.todo.todo.repository.RoleRepository;
+import com.todo.todo.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service(value = "userService")
+public class UserServiceImpl implements UserDetailsService, UserService
+{
+
+    @Autowired
+    private UserRepository userrepos;
+
+    @Autowired
+    private RoleRepository rolerepos;
+
+    @Transactional
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException
+    {
+        com.todo.todo.model.User user = userrepos.findByUsername(username);
+        if (user == null)
+        {
+            throw new UsernameNotFoundException("Invalid username or password.");
+        }
+        return new org.springframework.security.core.userdetails.User(user.getUsername(), user.getPassword(), user.getAuthority());
+    }
+
+    public com.todo.todo.model.User findUserById(long id) throws EntityNotFoundException
+    {
+        return userrepos.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Long.toString(id)));
+    }
+
+    public List<com.todo.todo.model.User> findAll()
+    {
+        List<com.todo.todo.model.User> list = new ArrayList<>();
+        userrepos.findAll().iterator().forEachRemaining(list::add);
+        return list;
+    }
+
+    @Override
+    public void delete(long id)
+    {
+        if (userrepos.findById(id).isPresent())
+        {
+            userrepos.deleteById(id);
+        }
+        else
+        {
+            throw new EntityNotFoundException(Long.toString(id));
+        }
+    }
+
+    @Transactional
+    @Override
+    public com.todo.todo.model.User save(com.todo.todo.model.User user)
+    {
+        com.todo.todo.model.User newUser = new User();
+        newUser.setUsername(user.getUsername());
+        newUser.setPasswordNoEncrypt(user.getPassword());
+
+        ArrayList<UserRoles> newRoles = new ArrayList<>();
+        for (UserRoles ur : user.getUserRoles())
+        {
+            newRoles.add(new UserRoles(newUser, ur.getRole()));
+        }
+        newUser.setUserRoles(newRoles);
+
+        for (Todo q : user.getTodos())
+        {
+            newUser.getTodos().add( new Todo(q.getTodo(), newUser));
```

Comment:
It looks like the todo objects don't have a getTodo() method. Also, you are missing something in your constructor. Make sure what you're passing in here matches what it expects in the todo model!